lib/src/components/NxSearchDropdown/__tests__/NxSearchDropdown.test.tsx

```diff
@@ -4,225 +4,203 @@
+
+    expect(quickRender({ disabled: true, matches: [{ id: '1', displayName: '1' }]}).getByRole('searchbox'))
+        .toBeDisabled();
+
+    let el = quickRender({ disabled: true, matches: [{ id: '1', displayName: '1' }]});
+    expect(el.getByRole('menuitem', { name: /1/i, hidden: true })).toBeDisabled();
 
-    expect(getShallow({ long: true }).find(NxFilterInput)).toHaveClassName('nx-text-input--long');
+    el = quickRender({ disabled: false, matches: [{ id: '1', displayName: '1' }]});
+    expect(el.getByRole('menuitem', { name: /1/i, hidden: true })).not.toBeDisabled();
   });
 
-  it('passes the disabled prop to the input and buttons', function() {
-    expect(getShallow().find(NxFilterInput)).toHaveProp('disabled', undefined);
-    expect(getShallow({ disabled: undefined }).find(NxFilterInput)).toHaveProp('disabled', undefined);
-    expect(getShallow({ disabled: null }).find(NxFilterInput)).toHaveProp('disabled', undefined);
-    expect(getShallow({ disabled: false }).find(NxFilterInput)).toHaveProp('disabled', undefined);
+  it('renders dropdown with role="alert" and empty text if there are no matching elements', function() {
+    const el = quickRender({ searchText: 'foo', matches: []});
 
-    expect(getShallow({ disabled: true }).find(NxFilterInput)).toHaveProp('disabled', true);
+    expect(el.getByRole('alert')).toBeInTheDocument();
+    expect(el.getByRole('alert')).toHaveTextContent('No Results Found');
+  });
+
+  it('sets the empty message from the emptyMessage prop', function() {
+    const el = quickRender({ searchText: 'foo', emptyMessage: 'asdfasdf' });
 
-    expect(getShallow({ disabled: true, matches: [{ id: '1', displayName: '1' }] }).find('.nx-dropdown-button'))
-        .toHaveProp('disabled', true);
-    expect(getShallow({ disabled: false, matches: [{ id: '1', displayName: '1' }] }).find('.nx-dropdown-button'))
-        .toHaveProp('disabled', undefined);
+    expect(el.getByRole('alert')).toHaveTextContent('asdfasdf');
   });
 
-  it('renders an NxDropdownMenu', function() {
-    expect(getShallow()).toContainMatchingElement(NxDropdownMenu);
-    expect(getShallow({ searchText: 'foo', disabled: true })).toContainMatchingElement(NxDropdownMenu);
-    expect(getShallow({ searchText: 'foo', disabled: false })).toContainMatchingElement(NxDropdownMenu);
-    expect(getShallow({ searchText: 'foo', disabled: undefined })).toContainMatchingElement(NxDropdownMenu);
-    expect(getShallow({ searchText: 'foo', disabled: null })).toContainMatchingElement(NxDropdownMenu);
-    expect(getShallow({ searchText: 'foo' })).toContainMatchingElement(NxDropdownMenu);
+  it('renders dropdown with role="menu" if there are matching elements', function() {
+    const el = quickRender({ searchText: 'foo', matches: [{ id: '1', displayName: '1' }]});
+
+    expect(el.getByRole('menu')).toBeInTheDocument();
   });
 
   it('sets aria-hidden to true on the dropdown menu if the searchText is empty or disabled is true', function() {
```

Comment:
I'm not sure it's the most useful to have RTL search for the hidden menu.  Better maybe to just assert that there is no non-hidden menu in these cases

Reply:
> I'm not sure it's the most useful to have RTL search for the hidden menu. Better maybe to just assert that there is no non-hidden menu in these cases

Changed here: 00706381d6b2f3736a8182926ef36f1ec06d864e